cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-connect.php

```diff
@@ -290,23 +313,62 @@ public function setup() {
 		// Get the cloudinary url from plugin config.
 		$config = $this->plugin->config['settings']['connect'];
 		if ( ! empty( $config['cloudinary_url'] ) ) {
-
 			$this->config_from_url( $config['cloudinary_url'] );
-
 			$this->api = new Connect\Api( $this, $this->plugin->version );
-			$stats     = get_transient( '_cloudinary_usage' );
-			if ( empty( $stats ) ) {
-				// Get users plan.
-				$stats = $this->plugin->components['connect']->api->usage();
-				if ( ! is_wp_error( $stats ) && ! empty( $stats['media_limits'] ) ) {
-					$stats['max_image_size'] = $stats['media_limits']['image_max_size_bytes'];
-					$stats['max_video_size'] = $stats['media_limits']['video_max_size_bytes'];
-					set_transient( '_cloudinary_usage', $stats, HOUR_IN_SECONDS );
-				}
+			$this->usage_stats( true );
+		}
+	}
+
+	/**
+	 * Set the usage stats from the Cloudinary API.
+	 *
+	 * @param bool $refresh Flag to force a refresh.
+	 */
+	public function usage_stats( $refresh = false ) {
```

Comment:
@DavidCramer 

It seems that this method is never called to get stalled data. Because of that, it's always making a new request to the API to get fresh data.
Can you confirm that's the case?

Reply:
@pereirinha - You are right! I made the `$refresh` true to test. and left it in there.
I'll make the fix for it in the morning!

Reply:
@pereirinha - I removed the refresh flag, and changed the text as requested by the client.